app/leaderboard/page.tsx

```diff
@@ -94,18 +96,32 @@ export default function Page() {
     };
 
     setLoading(true);
+    setRankingdataloading(true);
     fetchLeaderboardToppersResult({
       addr: requestBody.addr,
       duration: timeFrameMap(duration),
     });
     fetchRankingResults(requestBody);
     setLoading(false);
+    setRankingdataloading(false);
   }, [userAddress, status, apiCallDelay]);
 
   const fetchRankingResults = useCallback(
     async (requestBody: LeaderboardRankingParams) => {
-      const response = await fetchLeaderboardRankings(requestBody);
-      setRanking(response);
+      try {
+        setRankingdataloading(true);
+      const response = await fetchLeaderboardRankings(requestBody);         
+        setRanking(response); 
+        setRankingdataloading(false);
+      
+      
+    } catch (error) {
+      console.error(error);
+    } finally {
+     
+     setRankingdataloading(false);
```

Comment:
we do false also twice here. This should be in catch since you handle the this line for the try block inside it here line number 115

Reply:
I can change that though

Reply:
i removed the finally block and handled it in the catch block

Reply:
i update the PR now with the modified changes